app/src/common/shared/com/igalia/wolvic/speech/SpeechServices.java

```diff
@@ -0,0 +1,54 @@
+package com.igalia.wolvic.speech;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.StringDef;
+
+import com.igalia.wolvic.BuildConfig;
+import com.igalia.wolvic.R;
+
+import java.lang.reflect.Constructor;
+
+public abstract class SpeechServices {
+    // Full names of the classes that implement the connection to each speech recognition service.
+    // These classes must implement the SpeechRecognizer interface and provide a constructor
+    // that takes a Context as its only parameter.
+    public static final String MEETKAI = "com.igalia.wolvic.speech.MKSpeechRecognizer";
+    public static final String HUAWEI_ASR = "com.igalia.wolvic.HVRSpeechRecognizer";
+
+    @StringDef(value = {MEETKAI, HUAWEI_ASR})
+    @interface Service {
+    }
+
+    public static final @Service
+    String DEFAULT = BuildConfig.SPEECH_SERVICES[0];
+
+    /**
+     * @param aContext
+     * @param service
+     * @return A new instance of the chosen speech recognition service.
+     * @throws ReflectiveOperationException
+     */
+    public static SpeechRecognizer getInstance(@NonNull Context aContext, @NonNull @Service String service)
+            throws ReflectiveOperationException {
+        Class<?> recognizerClass = Class.forName(service);
+        Constructor<?> constructor = recognizerClass.getConstructor(Context.class);
+        return (SpeechRecognizer) constructor.newInstance(new Object[]{aContext});
+    }
```

Comment:
We can not instantiate them normally because not all of the classes might be available at compilation time.

For example, this fails in the Oculus because `HVRSpeechRecognizer` is not in the class path:

```java
case MEETKAI:
    return new MKSpeechRecognizer(aContext);   // works
case HUAWEI_ASR:
    return new HVRSpeechRecognizer(aContext);   // error: cannot find symbol
```

I have been looking for an alternative but could not find anything better than using reflection.